pkg/gitnessclient/gitness._types.go

```diff
@@ -0,0 +1,170 @@
+	} `json:"author"`
+	Committer struct {
+		Identity struct {
+			Name  string `json:"name"`
+			Email string `json:"email"`
+		} `json:"identity"`
+		When time.Time `json:"when"`
+	} `json:"committer"`
+	Stats struct {
+		Total struct {
+			Insertions int `json:"insertions"`
+			Deletions  int `json:"deletions"`
+			Changes    int `json:"changes"`
+		} `json:"total"`
+	} `json:"stats"`
+}
+type CommitsResponse struct {
+	Commits []Commit `json:"commits"`
+}
+
+type PullRequest struct {
+	Number           int     `json:"number"`
+	Created          int64   `json:"created"`
+	Edited           int64   `json:"edited"`
+	State            string  `json:"state"`
+	IsDraft          bool    `json:"is_draft"`
+	Title            string  `json:"title"`
+	Description      string  `json:"description"`
+	SourceRepoID     int     `json:"source_repo_id"`
+	SourceBranch     string  `json:"source_branch"`
+	SourceSha        string  `json:"source_sha"`
+	TargetRepoID     int     `json:"target_repo_id"`
+	TargetBranch     string  `json:"target_branch"`
+	Merged           *bool   `json:"merged"`
+	MergeMethod      *string `json:"merge_method"`
+	MergeCheckStatus string  `json:"merge_check_status"`
+	MergeTargetSha   string  `json:"merge_target_sha"`
+	MergeBaseSha     string  `json:"merge_base_sha"`
+	Author           struct {
+		ID          int    `json:"id"`
+		UID         string `json:"uid"`
+		DisplayName string `json:"display_name"`
+		Email       string `json:"email"`
+		Type        string `json:"type"`
+		Created     int64  `json:"created"`
+		Updated     int64  `json:"updated"`
+	} `json:"author"`
+	Merger *struct {
+		ID          int    `json:"id"`
+		UID         string `json:"uid"`
+		DisplayName string `json:"display_name"`
+		Email       string `json:"email"`
+		Type        string `json:"type"`
+		Created     int64  `json:"created"`
+		Updated     int64  `json:"updated"`
+	} `json:"merger"`
+	Stats struct {
+		Commits      int `json:"commits"`
+		FilesChanged int `json:"files_changed"`
+	} `json:"stats"`
+}
+
+type ApiRepository struct {
+	Id             int    `json:"id"`
+	ParentId       int    `json:"parent_id"`
+	Identifier     string `json:"identifier"`
+	Path           string `json:"path"`
+	Description    string `json:"description"`
+	IsPublic       bool   `json:"is_public"`
+	CreatedBy      int    `json:"created_by"`
+	Created        int64  `json:"created"`
+	Updated        int64  `json:"updated"`
+	Size           int    `json:"size"`
+	SizeUpdated    int    `json:"size_updated"`
+	DefaultBranch  string `json:"default_branch"`
+	ForkId         int    `json:"fork_id"`
+	NumForks       int    `json:"num_forks"`
+	NumPulls       int    `json:"num_pulls"`
+	NumClosedPulls int    `json:"num_closed_pulls"`
+	NumOpenPulls   int    `json:"num_open_pulls"`
+	NumMergedPulls int    `json:"num_merged_pulls"`
+	Importing      bool   `json:"importing"`
+	GitUrl         string `json:"git_url"`
+	Uid            string `json:"uid"`
+}
+
+type StaticContext struct {
```

Comment:
This type is unnecessary or should be named differently. The `gitprovider` package already has the `StaticGitContext` type.

Reply:
yes but we can use types from `gitprovider` as we have discussed previously

Reply:
We didn't discuss this. Take a look at how other github clients work, they do not have dependencies to our types. All the methods only support arguments that the method uses.

Reply:
You are right i am updating changes.